### ISSUE TITLE
Umbrella issue for Katacoda removal

### ISSUE THREAD

**Issue by reylejano** [kubernetes/website#33936]
## Problem
(The public part of) Katacoda shut down on June 15, 2022 
https://www.oreilly.com/online-learning/leveraging-katacoda-technology.html

The remaining part of Katacoda, that Kubernetes uses, is due to shut down ~late 2022~ early 2023.

Related to https://github.com/kubernetes/website/issues/33918 and https://github.com/kubernetes/website/issues/38785

## Discussion

**Use the [GitHub Discussion](https://github.com/kubernetes/website/discussions/38878) to discuss what we should replace Katacoda with.**

## Specific steps

SIG Docs members propose to edit pages that use Katacoda to mark that the sandbox is unavailable. When an alternative is available then update the affected pages to use the alternative.

**Pages to Update:**
https://kubernetes.io/docs/tutorials/hello-minikube/
https://kubernetes.io/docs/tutorials/kubernetes-basics/
https://kubernetes.io/docs/tutorials/kubernetes-basics/create-cluster/cluster-intro/
https://kubernetes.io/docs/tutorials/kubernetes-basics/create-cluster/cluster-interactive/
https://kubernetes.io/docs/tutorials/kubernetes-basics/deploy-app/
https://kubernetes.io/docs/tutorials/kubernetes-basics/deploy-app/deploy-intro/
https://kubernetes.io/docs/tutorials/kubernetes-basics/deploy-app/deploy-interactive/
https://kubernetes.io/docs/tutorials/kubernetes-basics/explore/
https://kubernetes.io/docs/tutorials/kubernetes-basics/explore/explore-intro/
https://kubernetes.io/docs/tutorials/kubernetes-basics/explore/explore-interactive/
https://kubernetes.io/docs/tutorials/kubernetes-basics/expose/
https://kubernetes.io/docs/tutorials/kubernetes-basics/expose/expose-intro/
https://kubernetes.io/docs/tutorials/kubernetes-basics/expose/expose-interactive/
https://kubernetes.io/docs/tutorials/kubernetes-basics/scale/
https://kubernetes.io/docs/tutorials/kubernetes-basics/scale/scale-intro/
https://kubernetes.io/docs/tutorials/kubernetes-basics/scale/scale-interactive/
https://kubernetes.io/docs/tutorials/kubernetes-basics/update/
https://kubernetes.io/docs/tutorials/kubernetes-basics/update/update-intro/
https://kubernetes.io/docs/tutorials/kubernetes-basics/update/update-interactive/
https://kubernetes.io/docs/tutorials/configuration/
https://kubernetes.io/docs/tutorials/configuration/configure-java-microservice/
https://kubernetes.io/docs/tutorials/configuration/configure-java-microservice/configure-java-microservice/
https://kubernetes.io/docs/tutorials/configuration/configure-java-microservice/configure-java-microservice-interactive/
https://kubernetes.io/docs/tutorials/configuration/configure-redis-using-configmap/
https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/#version-removal

## Additional Information

/remove-kind bug
/priority important-soon

**Comment by Debanitrkl** [comment]
https://github.com/kubernetes-sigs/contributor-katacoda
It's the place where we work on katacoda at Contribex.

**Comment by NitishKumar06** [comment]
Hey @reylejano Do we have to all these pages you've mentioned in the Pages to Update

**Comment by shannonxtreme** [comment]
> @Debanitrkl mentioned on [Slack](https://kubernetes.slack.com/archives/C1J0BPD2M/p1653459219238159?thread_ts=1653458820.939779&cid=C1J0BPD2M) that SIG ContribEx is looking at [killercoda](https://killercoda.com/creators)
> 
> @shannonxtreme [mentioned](https://kubernetes.slack.com/archives/C1J0BPD2M/p1653477524404569?thread_ts=1653458820.939779&cid=C1J0BPD2M) turning the katacoda tutorials to use minikube
> 
> Do we as SIG Docs want to continue to maintain these tutorials if they use killercoda or minikube?
> Should we turn the tutorials into instructions to use on any Kubernetes cluster?

Imo we can direct users to playgrounds or minikube or whatever (maybe a standard before you begin that says set up a Kubernetes cluster using an option such as X, Y, or Z) and keep our responsibilities scoped to platform agnostic instructions. 

I just enjoyed the interactive tutorials, but maintaining their compatibility and maintenance with platform X shouldn't be on SIG Docs

**Comment by afbjorklund** [comment]
As far as I know, the katacoda examples _are_ using minikube at the moment ?

An older version (1.18 vs 1.25), and run from a control plane terminal, but anyway.

**Comment by shannonxtreme** [comment]
Yeah they are using minikube so we'd basically be directing the user to use minikube wherever they want, instead of using the katacoda playground. It'll be less directly _interactive_ sadly

**Comment by afbjorklund** [comment]
I added some notes and links, on how to improve the minikube "none" driver for this scenario.

Providing the virtual machine and the web console for it would be up to the "learning platform".

**Comment by afbjorklund** [comment]
The jupyter notebooks are quite nice.

https://github.com/kubernetes-client/python/blob/master/examples/notebooks/README.md

Unfortunately they use python, not kubectl...

https://github.com/kubernetes-client/python/blob/master/examples/notebooks/create_deployment.ipynb

**Comment by sftim** [comment]
I suggest that we change the page templates so that any page using a Katacoda shortcode gets a removal warning, to explain that Katacoda is shutting down.

https://gohugo.io/templates/shortcode-templates/#checking-for-existence explains how to write a conditional based on the presence of a shortcode.

**Comment by bwplotka** [comment]
FYI: We used Katacoda heavily for the Thanos project, and we discussed some alternatives to consider - but definitely not in so short timespan: https://github.com/thanos-io/thanos/issues/5385

TL;DR https://killercoda.com/ (mentioned already) or https://instruqt.com/

**Comment by adamwitwer** [comment]
Hi team. I run the product team at O'Reilly, and we've set up a separate server for Kubernetes.io. The Katacoda scenarios on those pages will not be affected by the shutdown of the public site. My apologies for the miscommunication. @reylejano I've sent you an email with more context.

**Comment by sftim** [comment]
> Does that mean even [this](https://katacoda.com/k8scontributors) is supported, because it also comes under Kubernetes upstream but not directly linked to kubernetes.io

https://katacoda.com/k8scontributors will not survive the shutdown of https://katacoda.com/ and _does_ need to be migrated. I recommend @Debanitrkl that you file a separate issue elsewhere, for SIG ContribEx to track.

**Comment by sftim** [comment]
/remove-priority important-soon

We need to decide what priority to set for this.

**Comment by utkarsh-singh1** [comment]
Hi, i think this https://kubernetes.io/docs/tasks/job/automated-tasks-with-cron-jobs/ is skipped in the above list. just found this while going through the docs.

**Comment by RichardoC** [comment]
Is it worth migrating these to https://killercoda.com/ ? Given it's the organisation that already host the CKA/CKS mock exams it might be a good fit?

**Comment by sftim** [comment]
> Is it worth migrating these to https://killercoda.com/ ?

There's two ways to consider that question:
- is it beneficial to readers to do that migration?
- is there some other work that SIG Docs should reasonably prioritise over doing that migration?

My feeling is that the answers are Yes and Yes respectively. Migrating to something that's not Katacoda _might_ help readers, even with the need to sign up, but the effort it'd require is best spent elsewhere (I have a [query](https://github.com/kubernetes/website/issues?q=is%3Aissue+is%3Aopen+label%3Apriority%2Fcritical-urgent%2Cpriority%2Fimportant-soon%2Cpriority%2Fimportant-longterm+sort%3Acreated-asc) to illustrate).

**Comment by utkarsh-singh1** [comment]
Hi, I think this https://kubernetes.io/docs/contribute/style/style-guide/#katacoda-embedded-live-environment is skipped in the above list. just found this while going through the docs.

**Comment by sftim** [comment]
Also see https://github.com/kubernetes/website/issues/37817

**Comment by Ritikaa96** [comment]
[This](https://killercoda.com/creators/get-started) docs in killercoda says developer can use their katacoda repositories as well.
So they can  Import existing Katacoda scenarios and run them as killercoda scenarios too. It seems workable if this can be done. Anyone tried this? do we have any katacoda author to verify this?

**Comment by sftim** [comment]
> [This](https://killercoda.com/creators/get-started) docs in killercoda says developer can use their katacoda repositories as well. So they can Import existing Katacoda scenarios and run them as killercoda scenarios too. It seems workable if this can be done.

> Anyone tried this?

I don't think so.

> do we have any katacoda author to verify this?

The team behind Katacoda aren't likely to know the answer, to be honest: killercoda is a separate thing.

**Comment by wuestkamp** [comment]
It should be possible to port most scenarios from Katacoda to Killercoda, many have done it.
But there are also differences and there is a [Katacoda Migration Guide](https://wuestkamp.medium.com/katacoda-to-killercoda-migration-guide-d21961fc0c9b?source=friends_link&sk=6e93c2194234f703454e3eb1438712c9) for this

**Comment by Ritikaa96** [comment]
> to be honest: killercoda is a separate thing.

I agree with you there.
@wuestkamp I have seen this guide, but haven't tried or seen anyone successfully try this. if it is possible then using killercoda in k8s would also be feasible in all interactive tutorials.

**Comment by JFMeeks** [comment]
Hello, my name is Jesse Meeks and I am the product manager responsible for Katacoda here at O'Reilly and I am happy to put this team in touch with the Katacoda devs on our side in an effort to help you to migrate to Killercoda.

It is correct that the public facing Katacda.com site is not longer supported and it is our intent to take the site down entirely by the end of the year, though this is something Id like to discuss.

Kubernetes.io is the last remaining service running from the public Katacoda hosts and, in fact, has been seeing quite a bit of activity lately. This activity, while fantastic for Kubernetes.io, unfortunately has prompted some scrutiny and increased urgency around finalizing the shut down Katacoda.com. Because Katacoda will only be offered as a private service to O'Reilly customers upon shut down of the Katcoda.com site, there won't be a way for us to embed a publicly accessible Katacoda scenario outside of Oreilly.com, effectively breaking the terminal in your documentation.

As I mentioned, I'd like to be on the same page with regards to a shut down date, I certainly don't want to leave the team in a compromised position. If it is helpful to do so, I am happy to jump on a call or correspond via email: [jmeeks@oreilly.com](mailto:jmeeks@oreilly.com)

**Comment by AlanGreene** [comment]
@JFMeeks Has there been a change in approach since this comment in June that Kubernetes.io would not be affected by the public Katacoda shutdown? https://github.com/kubernetes/website/issues/33936#issuecomment-1145151104
> Hi team. I run the product team at O'Reilly, and we've set up a separate server for Kubernetes.io. The Katacoda scenarios on those pages will not be affected by the shutdown of the public site.

Perhaps there's been additional communication since then that's not reflected in this issue.

**Comment by sftim** [comment]
https://github.com/kubernetes/website/issues/33936#issuecomment-1346832443 was originally posed as https://github.com/kubernetes/website/issues/37817#issuecomment-1341412180; however, _this_ is the better issue to track the removal, now it's confirmed as something we must do.

**Comment by nikitar** [comment]
I inquired with killercoda about minikube support, but the bigger question is where to migrate in general. I'm happy to help with the work, but I'm not a maintainer, so I'm not really in position to pick the destination. And I imagine there are standard processes (RFC?) to make those sorts of decisions.

**Comment by nikitar** [comment]
@sftim @afbjorklund What's the process for making those sorts of decisions?

Killercoda folks have no plans of adding minikube to their k8s images, but if it's really necessary they suggest using the ubuntu image and installing minikube manually. Presumably it'd look something like

```
apt install virtualbox
curl -sL https://path/to/binary
```

**Comment by afbjorklund** [comment]
It is not necessary, as we have debated elsewhere **all** of minikube and kind and kubeadm are valid ways of installing k8s.

You don't _need_ VirtualBox to run minikube, it is _one_ of the original drivers but it is not even the only hypervisor anymore...

----

The main goal of the "machine" provisioning is to create a machine and **provision** (install) a container runtime, without having to use a proprietary tool like Docker Desktop. If you already have one, you can instead choose to run nested containers (like kind). Finally, once the machine (nodes) is up and running it will **bootstrap** up the cluster (with kubeadm). The rest is just lifecycle tools (including autopause), and convenience wrappers for features such as image or dashboard.

If you want to do your own provisioning and bootstrapping, then you don't _have_ to use `minikube` but invent your own.

**Comment by afbjorklund** [comment]
> but if it's really necessary they suggest using the ubuntu image and installing minikube manually

you don't have to install virtualbox, but you do have to install all the _other_ requirements for it to work

*  /etc/modules-load.d/k8s.conf, /etc/sysctl.d/k8s.conf
* iptables (>= 1.4.21), iproute2, socat, util-linux, mount, ebtables, ethtool, conntrack
* cri-tools
* cni-plugins
* docker
* cri-dockerd

Basically it comes to down to installing all requirements for kubeadm, and also preloading the images ?

Naturally you also have do all the _other_ cluster configuration, such as CRI and CNI and untainting master

**Comment by nikitar** [comment]
@afbjorklund Thanks for the information. I'm gonna hold off on looking farther into this until there is a decision on whether/where to migrate the tutorials. (Would love to understand how those kinds of decisions are made, I assume one of the SIGs would be involved)

To be clear, killercoda does offer pre-configured k8s images too (see the 'Environments' section [here](https://killercoda.com/creators)), they just don't include minikube. kubernetes-kubeadm-1node can be used for all tutorials except the minikube one, I imagine.

**Comment by afbjorklund** [comment]
> I'm gonna hold off on looking farther into this until there is a decision on whether/where to migrate the tutorials. 

I added some new bug reports, on how to improve the appearance of the minikube "none" driver...

* https://github.com/kubernetes/minikube/issues/15576

* https://github.com/kubernetes/minikube/issues/15577

Even if it ends up **not** being used for the kubernetes online tutorials anymore (😿), it **is** still a bit broken.


----

Currently it is a bit indecisive about installing on the node (like kubeadm), or in nested containers (like kind)
But that seems to apply to _all_ environments, even though running on the control plane is an anti-pattern ?

But in a cloud context, it would be more expensive to have 1 VM for the "laptop" and 1 VM for the "cluster"

The main reason why dind is cheaper, is because we are cheating kubernetes about the resources available.
It thinks that there are two nodes available, but in reality they are both _sharing_ the cpus and memory (and disk).

Also it is horribly complex to maintain, and seems to cause issues not found elsewhere, but that's another story.

> kubernetes-kubeadm-1node can be used for all tutorials except the minikube one, I imagine.

I don't really see how you can (easilly) run the same environment "at home" (offline), though. 🤔

Can see the markdown, but I don't know where the images are created and coming from... ?

https://killercoda.com/kubernetes

https://github.com/killercoda/scenarios-kubernetes

Supposedly you "only" have to install ubuntu and use the kubeadm/k3s documentation to install.

This is what is being done for [Lima](https://lima-vm.io/), though it is still struggling a bit with the multi-node scenario.

`limactl start template://k8s` [examples/k8s.yaml](https://github.com/lima-vm/lima/blob/master/examples/k8s.yaml)

`limactl start template://k3s` [examples/k3s.yaml](https://github.com/lima-vm/lima/blob/master/examples/k3s.yaml)

----

There are a _lot_ of different options, but so far none that uses `kind` / `k3d` (from what I can tell) ?

`kubernetes-kubeadm-1node`

`kubernetes-kubeadm-2nodes`

`kubernetes-k3s-1node`

`kubernetes-k3s-2nodes`

But I guess you can start with the docker image, and then use the documentation to install them ?

`limactl start --name ubuntu template://docker-rootful` (docker)

`limactl start --name ubuntu template://podman-rootful` (podman)

(I made some kind.yaml and k3d.yaml, but nobody really wants them)

katacoda: `ubuntu` (docker + podman)

So far no support for "nerdctl", which is what one would want to talk with `containerd`/`buildkitd`
For instance if you want to be able to build images, without having to save/load or push/pull them...

That is the main reason minikube is still defaulting to `dockerd` (moby), to not have to run two engines.
And for quick experiments, it can be very useful to just be able to run a development container locally.

----

As far as I know, it is up to SIG Docs to find hosting for the tutorials - and killercoda looks like the target

I was just trying to improve the content, so that you don't run software from 2021 instead of from 2023...

` version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.2", GitCommit:"faecb196815e248d3ecfb03c680a4507229c2a56", GitTreeState:"clean", BuildDate:"2021-01-13T13:28:09Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}`


Our end goal is the same: https://kubernetes.io/blog/2016/07/minikube-easily-run-kubernetes-locally/

**Comment by sftim** [comment]
I'd like to announce the removal to end users before the switch off happens. See issue #38785.

**Comment by sftim** [comment]
This issue is relevant to https://github.com/kubernetes/website/pull/38744

**Comment by afbjorklund** [comment]
@nikitar **did you have any other alternatives for migration hosting, besides killercoda ?**

> kubernetes-kubeadm-1node can be used for all tutorials except the minikube one, I imagine.

This seems accurate, and there is no longer a need to run `start.sh` or `launch.sh` for the other tutorials:

https://itnext.io/katacoda-to-killercoda-migration-guide-d21961fc0c9b

"On Katacoda it was (sometimes) necessary to run a magical `launch.sh` to get that Kubernetes environment running. This is no longer necessary on Killercoda because all K8s environments are always running and ready!"

I am not sure what the side effects are, when they bypass the 2 vCPU requirement from `kubeadm` ?

```
        [ERROR NumCPU]: the number of available CPUs 1 is less than the required 2
```

I don't know if it is possible to access the kubernetes dashboard, but it seemed like a work-in-progress ?

https://killercoda.com/examples/scenario/kubernetes-dashboard shows how to do it "The Hard way"

----

If you want to continue to support `hello minikube`, then that _could_ be done with the `ubuntu` image.

https://kubernetes.io/docs/tutorials/hello-minikube/

https://killercoda.com/examples/scenario/ubuntu-simple

It would need some fixes, such as the bugs mentioned above or running with smaller amount of resources.

```
⛔  Exiting due to RSRC_INSUFFICIENT_CORES: Requested cpu count 2 is greater than the available cpus of 1
```

But for the rest of the scenarios, there should be nothing blocking the katacoda to killercoda conversion.

**Comment by afbjorklund** [comment]
I made a patched version of `minikube` (for NumCPU), and using this version it runs on killercoda:

```console
ubuntu $ minikube version
minikube version: v1.28.0
commit: b45c13b3eb399ba7c63db31609f4a013b1f8d638
ubuntu $ minikube start --driver=none --force
😄  minikube v1.28.0 on Ubuntu 20.04 (amd64)
❗  minikube skips various validations when --force is supplied; this may lead to unexpected behavior
✨  Using the none driver based on existing profile

⛔  Requested cpu count 2 is greater than the available cpus of 1


⛔  None has less than 2 CPUs available, but Kubernetes requires at least 2 to be available


⛔  Requested cpu count 2 is greater than the available cpus of 1


⛔  None has less than 2 CPUs available, but Kubernetes requires at least 2 to be available


🧯  The requested memory allocation of 1983MiB does not leave room for system overhead (total system memory: 1983MiB). You may face stability issues.
💡  Suggestion: Start minikube with less memory allocated: 'minikube start --memory=1983mb'

👍  Starting control plane node minikube in cluster minikube
🔄  Restarting existing none bare metal machine for "minikube" ...
ℹ️  OS release is Ubuntu 20.04.5 LTS
🐳  Preparing Kubernetes v1.25.3 on Docker 20.10.12 ...
    ▪ kubelet.resolv-conf=/run/systemd/resolve/resolv.conf
    ▪ Generating certificates and keys ...
    ▪ Booting up control plane ...
    ▪ Configuring RBAC rules ...
🤹  Configuring local host environment ...

❗  The 'none' driver is designed for experts who need to integrate with an existing VM
💡  Most users should use the newer 'docker' driver instead, which does not require root!
📘  For more information, see: https://minikube.sigs.k8s.io/docs/reference/drivers/none/

❗  kubectl and minikube configuration will be stored in /root
❗  To use kubectl or minikube commands as your own user, you may need to relocate them. For example, to overwrite your own settings, run:

    ▪ sudo mv /root/.kube /root/.minikube $HOME
    ▪ sudo chown -R $USER $HOME/.kube $HOME/.minikube

💡  This can also be done automatically by setting the env var CHANGE_MINIKUBE_NONE_USER=true
🔎  Verifying Kubernetes components...
    ▪ Using image gcr.io/k8s-minikube/storage-provisioner:v5
🌟  Enabled addons: default-storageclass, storage-provisioner
💡  kubectl not found. If you need it, try: 'minikube kubectl -- get pods -A'
🏄  Done! kubectl is now configured to use "minikube" cluster and "default" namespace by default
ubuntu $ minikube kubectl version
WARNING: This version information is deprecated and will be replaced with the output from kubectl version --short.  Use --output=yaml|json to get the full version.
Client Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.3", GitCommit:"434bfd82814af038ad94d62ebe59b133fcb50506", GitTreeState:"clean", BuildDate:"2022-10-12T10:57:26Z", GoVersion:"go1.19.2", Compiler:"gc", Platform:"linux/amd64"}
Kustomize Version: v4.5.7
Server Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.3", GitCommit:"434bfd82814af038ad94d62ebe59b133fcb50506", GitTreeState:"clean", BuildDate:"2022-10-12T10:49:09Z", GoVersion:"go1.19.2", Compiler:"gc", Platform:"linux/amd64"}
ubuntu $ minikube kubectl -- get nodes -o wide
NAME     STATUS     ROLES           AGE   VERSION   INTERNAL-IP   EXTERNAL-IP   OS-IMAGE             KERNEL-VERSION      CONTAINER-RUNTIME
ubuntu   NotReady   control-plane   50s   v1.25.3   172.30.1.2    <none>        Ubuntu 20.04.5 LTS   5.4.0-131-generic   docker://20.10.12
```

There is of course a large number of warnings, and you do have to install some extra dependencies:

```console
sudo apt update
sudo apt install -y conntrack socat

curl -LO https://github.com/Mirantis/cri-dockerd/releases/download/v0.3.0/cri-dockerd_0.3.0.3-0.ubuntu-focal_amd64.deb
sudo dpkg -i cri-dockerd_0.3.0.3-0.ubuntu-focal_amd64.deb
```

But cri-tools and cni-plugins have already been installed, from the deprecated kubic repositories.

```
cri-tools/now 1.21.0~2 amd64 [installed,local]
containernetworking-plugins/now 100:1.1.1~1 amd64 [installed,local]
```

**Comment by sftim** [comment]
@reylejano I updated the issue description to highlight the new GitHub discussion.

**Comment by afbjorklund** [comment]
I would suggest to **remove** minikube, also from the first scenario (all other scenarios start with it running).

https://kubernetes.io/docs/tutorials/kubernetes-basics/create-cluster/ - call it "Connect to Cluster" or such ?

This will allow migrating to killercoda, and leaving "Hello, Minikube" as the **only** page needing `minikube`.

* https://github.com/kubernetes/minikube/issues/15651#issuecomment-1383172188

---

Then, if someone wants to contribute scenarios on how to install and run minikube or kind "from scratch"...

But those can probably refer to the project documentation, and link straight over to [minikube](https://minikube.sigs.k8s.io/) and [kind](https://kind.sigs.k8s.io/) sites ?

The killercoda environment comes with a pre-installed and running kubeadm environment, without `start.sh`

For the user, this means they can start directly with `kubectl` without having to first look at the cluster installer.

**Comment by NitishKumar06** [comment]
> I would suggest to **remove** minikube, also from the first scenario (all other scenarios start with it running).
> 
> https://kubernetes.io/docs/tutorials/kubernetes-basics/create-cluster/ - call it "Connect to Cluster" or such ?
> 
> This will allow migrating to killercoda, and leaving "Hello, Minikube" as the **only** page needing `minikube`.
> 
> * [Outdated description of minikube in upstream documentation minikube#15651 (comment)](https://github.com/kubernetes/minikube/issues/15651#issuecomment-1383172188)
> 
> Then, if someone wants to contribute scenarios on how to install and run minikube or kind "from scratch"...
> 
> But those can probably refer to the project documentation, and link straight over to [minikube](https://minikube.sigs.k8s.io/) and [kind](https://kind.sigs.k8s.io/) sites ?
> 
> The killercoda environment comes with a pre-installed and running kubeadm environment, without `start.sh`
> 
> For the user, this means they can start directly with `kubectl` without having to first look at the cluster installer.

That's good approach! However, I'm likely unsure if sig-docs is maintaining the minikube documentation. How would it sound to redirect an important topic link to documentation outside the Kubernetes docs :( ?

**Comment by afbjorklund** [comment]
> However, I'm likely unsure if sig-docs is maintaining the minikube documentation.

We are talking about the "[Hello, Minikube](https://kubernetes.io/docs/tutorials/hello-minikube/)" page for Katacoda here, not the [minikube documentation](https://minikube.sigs.k8s.io/docs/)...
Just saying that when the online tutorial is removed (with Katacoda), there will not be so much left of it ?

So it would then look more like the information on this page: https://kubernetes.io/docs/tasks/tools/
A short introduction, and then a link with details. Already happens, with e.g. the container runtimes ?

It's fine to keep the separate page of course, but then it needs to be kept in sync with the project.

And the k8s.io front page today, skips right over it - and goes straight to "[Learn Kubernetes Basics](https://kubernetes.io/docs/tutorials/kubernetes-basics/)"

**Comment by sftim** [comment]
> This will allow migrating to killercoda

**Please use https://github.com/kubernetes/website/discussions/38878 to discuss what should replace Katacoda** - we haven't yet decided.

**Comment by sftim** [comment]
I'm proposing that we work towards removing Katacoda by the end of March 2023. How does that sound, Kubernetes folks?

**Comment by sftim** [comment]
Banner to announce that: https://github.com/kubernetes/website/pull/39257

**Comment by tengqm** [comment]
Let's get this done asap. We keep receiving complaints about Katacoda not working as documented. There are simply many issues reported none gets addressed in time.

**Comment by sftim** [comment]
I've contacted O'Reilly with our proposed shutdown date (end of March 2023).

**Comment by medyagh** [comment]
I purpose to continue using minikube in the future implementation, (for example https://killercoda.com/examples/scenario/kubernetes-1node)  since minikube provides special features for local development (such as fast "image build" and "dashboard" and mount, ...)  , that would be more realistic way of helping user in their journey so they could continue it on their local machine as well. 

most users are not interested to bootstrap using kubeadm but they are more interested in building/accessing their apps.

**Comment by sftim** [comment]
The best place to propose how we replace Katacoda is in the [discussion](https://github.com/kubernetes/website/discussions/38878) that we're using.

**Comment by medyagh** [comment]
Hello regarding the katakoda shutdown
https://kubernetes.io/blog/2023/02/14/kubernetes-katacoda-tutorials-stop-from-2023-03-31/

I would like to request to convert the katakoda tutorials to a plain-text format to keep on kuberentes websites, these scenarios are very important for new users who just wanna try Kubernetes and at least we could save these tutorials that users have loved for years in a text format

**Comment by sftim** [comment]
@medyagh, the best place to propose that work is as a new feature request. Visit https://github.com/kubernetes/website/issues/new/choose to file one, and reference https://github.com/kubernetes/website/issues/33936#issuecomment-1479973661 when you do.

**Comment by sftim** [comment]
https://github.com/kubernetes/website/issues/40503 requests keeping https://kubernetes.io/docs/tutorials/configuration/configure-java-microservice/configure-java-microservice/

**Comment by sftim** [comment]
Let's close this, and open new issues for any snagging we encounter.

/close

**Comment by sftim** [comment]
Also see https://github.com/kubernetes/website/issues/42505
